privacyidea/lib/tokens/yubicotoken.py

```diff
@@ -138,6 +138,7 @@ def check_otp(self, anOtpVal, counter=None, window=None, options=None):
 
         apiId = get_from_config("yubico.id", DEFAULT_CLIENT_ID)
         apiKey = get_from_config("yubico.secret", DEFAULT_API_KEY)
+        YUBICO_URL = get_from_config("yubico.url", DEFAULT_YUBICO_URL)
```

Comment:
Cornelius Kölbel <notifications@github.com> writes:

>> @@ -138,6 +138,7 @@ def check_otp(self, anOtpVal, counter=None, window=None, options=None):
>>  
>>          apiId = get_from_config("yubico.id", DEFAULT_CLIENT_ID)
>>          apiKey = get_from_config("yubico.secret", DEFAULT_API_KEY)
>> +        YUBICO_URL = get_from_config("yubico.url", DEFAULT_YUBICO_URL)
>
> call this lowercase "yubico_url". But you need to change this in the function below.

Something like:

diff --git a/privacyidea/lib/tokens/yubicotoken.py b/privacyidea/lib/tokens/yubicotoken.py
index 7b69fe1..19e14d1 100644
--- a/privacyidea/lib/tokens/yubicotoken.py
+++ b/privacyidea/lib/tokens/yubicotoken.py
@@ -51,7 +51,7 @@ import binascii
 
 YUBICO_LEN_ID = 12
 YUBICO_LEN_OTP = 44
-YUBICO_URL = "http://api.yubico.com/wsapi/2.0/verify"
+YUBICO_URL = "https://api.yubico.com/wsapi/2.0/verify"
 DEFAULT_CLIENT_ID = 20771
 DEFAULT_API_KEY = "9iE9DRkPHQDJbAFFC31/dum5I54="
 
@@ -138,6 +138,7 @@ class YubicoTokenClass(TokenClass):
 
         apiId = get_from_config("yubico.id", DEFAULT_CLIENT_ID)
         apiKey = get_from_config("yubico.secret", DEFAULT_API_KEY)
+        yubico_url = get_from_config("yubico.url", YUBICO_URL)
 
         if apiKey == DEFAULT_API_KEY or apiId == DEFAULT_CLIENT_ID:
             log.warning("Usage of default apiKey or apiId not recomended!")
@@ -159,7 +160,7 @@ class YubicoTokenClass(TokenClass):
                  'id': apiId}
 
             try:
-                r = requests.post(YUBICO_URL,
+                r = requests.post(yubico_url,
                                   data=p)
 
                 if r.status_code == requests.codes.ok:

-- 
The only problem with troubleshooting is that the trouble shoots back.